### PR TITLE
Add `distribute releases edit` command

### PR DESCRIPTION
**Pull request by pfleidi**
This PR will add the `distribute releases edit` command which allows to enable or disable a release.

```
appcenter-cli (edit-release) » ./bin/appcenter.js distribute releases edit --help                                                                                                                            ~/work/microsoft/appcenter-cli

Enables or disables specified release

Usage: appcenter distribute releases edit -r|--release-id <arg> [-a|--app <arg>] <State>

Options:
    -r|--release-id <arg>             Release ID
    -a|--app <arg>                    Specify app in the <ownerName>/<appName> format
    State                             Release state
```

### CODE REVIEW

File: src/commands/distribute/releases/edit.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+@help("Enables or disables specified release")
+export default class EditReleaseCommand extends AppCommand {
+  @help("Release ID")
+  @shortName("r")
+  @longName("release-id")
+  @required
+  @hasArg
+  public releaseId: string;
+
+  @help("Release state")
```
Comment (owenniblock) [inline src/commands/distribute/releases/edit.ts]: I think the help text should display the valid states.

File: src/commands/distribute/releases/edit.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    let releaseDetails: models.ReleaseDetailsResponse;
+    try {
+      debug("Loading release details");
+      const httpRequest = await out.progress("Loading release details...", clientRequest<models.ReleaseDetailsResponse>(
+        (cb) => client.releases.getLatestByUser(this.releaseId, app.ownerName, app.appName, cb)
+      ));
+      if (httpRequest.response.statusCode >= 400) {
+        throw httpRequest.response.statusCode;
```
Comment (owenniblock) [inline src/commands/distribute/releases/edit.ts]: I'm not super keen on throwing a number like this - I'd rather handle the error cases here directly and only handle legitimate exceptions in the `catch` block.

File: test/commands/distribute/releases/edit-test.ts [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+      const result = await command.execute();
+
+      // Assert
+      expect((result as CommandFailedResult).errorMessage).to.eql(expectedErrorMessage);
+    });
+
+    it("if state option wasn't specified", async () => {
+      // Arrange
+      let errorMessage: string;
+      const expectedErrorMessage = "Missing required positional argument State";
```
Comment (owenniblock) [inline test/commands/distribute/releases/edit-test.ts]: Not your doing but a note to self - I really don't like the wording here. Using the terminology `required positional argument` seems a bit hard to understand to me.

File: src/commands/distribute/releases/edit.ts (hunk 2 above)
Comment (owenniblock) [reply src/commands/distribute/releases/edit.ts]: I see that we do this in other places too - still don't like it but if it's standard then don't worry about this one.

Comment (pfleidi) [reply src/commands/distribute/releases/edit.ts]: Yeah, I don't like it too much, either but I think this pattern was copied from another command which was using it.

Comment (owenniblock) [reply src/commands/distribute/releases/edit.ts]: Yeah, I noticed that after I submitted the review 😆 Let's worry about this another day!

Comment (hagould) [reply src/commands/distribute/releases/edit.ts]: we resolved it for this task! fair feedback

**Pull request merged**